### PR TITLE
Refactor sql readme.md

### PR DESCRIPTION
**Pull request by jaredmoo**
- Rename schema-* to package-pure-* in order to follow convention for python multi-version
- Rename package-2017-10-preview to package-composite-v1. The reasons for this are explained in package-2017-03-01-preview tag. The next tag that I add in the near future will be package-composite-v2, which will represent a breaking change & major version bump for clients.
- Improved description of various sections
- Whitespace/header consistency

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jaredmoo** [comment]
I would like review from the following SDK representatives:
- @marstr 
- @milismsft 
- @lmazuel 

Please note: I will soon be adding a new tag `package-composite-v2` which will be a major version bump due to changes to two of our most important APIs, `Database` and `ElasticPool`. The idea is that `package-composite-v2` and `package-composite-v1` can both be supported (for some time) moving forward in order to reduce pain of breaking changes.

**Comment by lmazuel** [comment]
@sarangan12 That's the perfect opportunity for you to check this PR contains the tag you need for ruby

File: specification/sql/resource-manager/readme.md [hunk 1]
```diff
@@ -4,42 +4,42 @@
...
+tag: package-composite-v1
 ```
 
-## Configuration for generating client SDKs
+## Composite packages
+
+The following packages may be composed from multiple api-versions.
 
-### Tag: package-2017-10-preview
+### Tag: package-composite-v1
```
Comment (lmazuel) [inline specification/sql/resource-manager/readme.md]: I love that tag :)

File: specification/sql/resource-manager/readme.md [hunk 2]
```diff
@@ -405,8 +449,7 @@ Please also specify `--go-sdk-folder=<path to the root directory of your azure-s
 output-folder: $(go-sdk-folder)/services/sql/mgmt/2014-04-01/sql
 ```
 
-
-## Python
```
Comment (lmazuel) [inline specification/sql/resource-manager/readme.md]: Please remove this Python section, there is currently two Python sections in this file and this one is the bad one.

**Review by lmazuel — CHANGES_REQUESTED** [review]
This is awesome, thanks Jared!
I see with this PR that for some reason Python has two sections. The second one should be removed. This is the only reason why I ask changes, this is really awesome!

File: specification/sql/resource-manager/readme.md [hunk 3]
```diff
@@ -369,16 +401,28 @@ go:
...
 ```
 
-### Tag: package-2017-03-preview and go
+#### Tag: package-pure-2017-10-preview and go
+
+These settings apply only when `--tag=package-2017-10-preview --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'package-pure-2017-10-preview' && $(go)
+output-folder: $(go-sdk-folder)/services/sql/mgmt/2017-103-01-preview/sql
```
Comment (marstr) [inline specification/sql/resource-manager/readme.md]: Typo here! Looks like the date is malformed.

File: specification/sql/resource-manager/readme.md [hunk 4]
```diff
@@ -4,42 +4,42 @@
...
 ### Basic Information
+
 These are the global settings for the Sql API.
 
 ``` yaml
 title: SqlManagementClient
 description: The Azure SQL Database management API provides a RESTful set of web services that interact with Azure SQL Database services to manage your databases. The API enables you to create, retrieve, update, and delete databases.
 openapi-type: arm
-tag: package-2017-03-preview
+tag: package-composite-v1
```
Comment (marstr) [inline specification/sql/resource-manager/readme.md]: I think this makes a lot of sense. It makes explicit the composite nature of input-file tags.

To be clear though, the `v1` here isn't something that we expect to be exposed to customers, and is just a hint for the SDK creator, right?

**Review by marstr — CHANGES_REQUESTED** [review]

File: specification/sql/resource-manager/readme.md (hunk 4 above)
Comment (jaredmoo) [reply specification/sql/resource-manager/readme.md]: Correct

**Review by marstr — APPROVED** [review]

**Comment by jaredmoo** [comment]
@payiAzure fyi

File: specification/sql/resource-manager/readme.md (hunk 2 above)
Comment (jaredmoo) [reply specification/sql/resource-manager/readme.md]: Done, thx

**Comment by jaredmoo** [comment]
@lmazuel I have removed the bad python section
@milismsft this change should work transparently for you, you can continue to use default tag (which is now `package-composite-v1`

Can we aim to get this merged today? thx

@trgrie fyi

**Review by lmazuel — APPROVED** [review]

**Review by milismsft — APPROVED** [review]

**Comment by jaredmoo** [comment]
@marstr  looks like we are good to merge, thx

**Pull request merged**